src/Xenial.Tasty.Cli/TastyServer.cs

```diff
@@ -0,0 +1,216 @@
+            await Task.CompletedTask;
+        }
+
+        internal async Task DoRequestCancellation()
+        {
+            CancellationRequested?.Invoke(this, EventArgs.Empty);
+            await Task.CompletedTask;
+        }
+
+        public Task Report(SerializableTestCase @case)
+            => ConsoleReporter.Report(@case);
+
+        public Task Report(IEnumerable<SerializableTestCase> @cases)
+           => ConsoleReporter.ReportSummary(@cases);
+
+        internal Action<IList<SerializableTastyCommand>>? CommandsRegistered;
+        internal Action? EndTestPipelineSignaled;
+        internal Action? TestPipelineCompletedSignaled = () => { };
+        public void RegisterCommands(IList<SerializableTastyCommand> commands)
+            => CommandsRegistered?.Invoke(commands);
+
+        public void SignalEndTestPipeline()
+            => EndTestPipelineSignaled?.Invoke();
+
+        public void SignalTestPipelineCompleted()
+            => TestPipelineCompletedSignaled?.Invoke();
+
+        public void ClearConsole()
+        {
+            try
+            {
+                Console.Clear();
+            }
+            catch (IOException) { /* Handle is invalid */}
+        }
+
+        public static class ConsoleReporter
+        {
+            public static ColorScheme Scheme = ColorScheme.Default;
+
+            static ConsoleReporter()
+                => Console.OutputEncoding = Encoding.UTF8;
+
+
+            static Lazy<int> SeparatorSize = new Lazy<int>(() =>
+            {
+                const int fallBackSize = 100;
+                try
+                {
+                    return Console.BufferWidth;
+                }
+                catch (IOException) { /* Handle is invalid */ }
+                return fallBackSize;
+            });
+
+            public static Task ReportSummary(IEnumerable<SerializableTestCase> tests)
+            {
+                var totalTests = tests.Count();
+                var failedTests = tests.Where(m => m.TestOutcome == TestOutcome.Failed).Count();
+                var ignoredTests = tests.Where(m => m.TestOutcome == TestOutcome.Ignored).Count();
+                var inconclusiveTests = tests.Where(m => m.TestOutcome == TestOutcome.NotRun).Count();
+                var successTests = tests.Where(m => m.TestOutcome == TestOutcome.Success).Count();
+                var outcome = tests.Where(t => t.TestOutcome > TestOutcome.Ignored).Min(t => t.TestOutcome);
+
+                var totalTime = tests.Sum(m => m.Duration);
+                var failedTime = tests.Where(m => m.TestOutcome == TestOutcome.Failed).Sum(m => m.Duration);
+                var ignoredTime = tests.Where(m => m.TestOutcome == TestOutcome.Ignored).Sum(m => m.Duration);
+                var inconclusiveTime = tests.Where(m => m.TestOutcome == TestOutcome.NotRun).Sum(m => m.Duration);
+                var successTime = tests.Where(m => m.TestOutcome == TestOutcome.Success).Sum(m => m.Duration);
```

Comment:
not very DRY

Reply:
Currently there is No way to have console reporters to *teleport* between assemblies. The signatur is also different. We need a more low level Implementation to *stream* console level output. I think we should a new issue, cause this is clearly out of the goal of this PR.